test/jdk/java/nio/Buffer/CharBufferAsCharSequenceTest.java

```diff
@@ -83,21 +83,33 @@ private static void addCases(String type, char[] buf, CharBuffer cb, List<Argume
         cases.add(Arguments.of(cb, buf, 0, buf.length, type + " full"));
 
         CharBuffer rndRange = randomizeRange(cb.duplicate());
-        cases.add(Arguments.of(rndRange, buf, rndRange.position(), rndRange.limit(), type + "  at " + rndRange.position() + " through " + rndRange.limit()));
+        cases.add(Arguments.of(rndRange, buf, rndRange.position(), rndRange.limit(), type + " at " + rndRange.position() + " through " + rndRange.limit()));
         cases.add(Arguments.of(rndRange.slice(), buf, rndRange.position(), rndRange.limit(), type + " sliced at " + rndRange.position() + " through " + rndRange.limit()));
 
         CharBuffer rndSlicedRange = randomizeRange(rndRange.slice());
-        cases.add(Arguments.of(rndSlicedRange, buf, rndRange.position() + rndSlicedRange.position(), rndRange.position() + rndSlicedRange.limit(), type + " sliced at " + rndRange.position() + " with position " + rndSlicedRange.position() + " and limit " + rndSlicedRange.limit()));
+        cases.add(Arguments.of(rndSlicedRange, buf, rndRange.position() + rndSlicedRange.position(),
+                rndRange.position() + rndSlicedRange.limit(), type + " sliced at " + rndRange.position()
+                        + " with position " + rndSlicedRange.position() + " and limit " + rndSlicedRange.limit()));
     }
 
+    /**
+     * Returns a {@code List} of {@link Arguments}, with each entry representing a test case scenario.
+     * <ul>
+     * <li>CharBuffer - the instance to be tested</li>
+     * <li>char[] - the data expected to be backing the current state of the CharBuffer</li>
+     * <li>int start - index (inclusive) into char[] where the CharBuffer should be positioned</li>
+     * <li>int stop - index (exclusive) into char[] where the CharBuffer should be limited</li>
+     * <li>String - description of the test scenario</li>
+     * </ul>
+     */
```

Comment:
That helps a bit but I think we need to break up charBufferArguments into simpler/maintainable methods that have clear comments to document the test cases that they generate. Right now it's just way too hard to see what is tested and not tested.

Reply:
If they duplicate the random tests, then can be removed.
Which cases would you remove?